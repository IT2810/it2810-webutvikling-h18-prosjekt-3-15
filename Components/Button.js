import React from "react";
import {TouchableOpacity} from "react-native";

class Button extends React.Component{
    constructor(props){
        super(props);
    }

    startTimer(event) {
        let time = (this.props.isMinutes) ? this.props.time * 60 : this.props.time;
        return this.props.startTimer(time)
    }

    render() {
        return (
            <TouchableOpacity onPress={this.startTimer}>
            </TouchableOpacity>
        )
    }
}

export default Button;