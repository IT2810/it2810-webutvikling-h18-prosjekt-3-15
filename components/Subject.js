import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
    AsyncStorage,
    Button,
} from 'react-native';
import {StackNavigator} from 'react-navigation';


export default class Subject extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            url: this.props.url
        }
    }

    render() {
        const nav = this.props.navigation;
        return (
            <View key={this.props.keyval} style={styles.subject}>
                <TouchableOpacity style={styles.subjectContainer}>
                    <Button title={this.props.subject} onPress={()=> nav.navigate('Subjects')}/>
                </TouchableOpacity>
                <TouchableOpacity onPress={this.props.deleteMethod} style={styles.subjectDelete}>
                    <Text style={styles.subjectDeleteText}>Delete</Text>
                </TouchableOpacity>
            </View>
        );
    }

}
const styles = StyleSheet.create({
    subject: {
        position: 'relative',
        padding: 40,
        paddingRight: 20,
        borderBottomWidth: 2,
        borderBottomColor: '#ededed'
    },
    subjectText: {
        paddingLeft: 20,
        borderLeftWidth: 10,
        borderLeftColor: '#4286f4',
        fontWeight: 'bold',
        opacity: 1,
    },
    subjectContainer: {
        position: 'absolute',
        justifyContent: 'center',
        alignSelf: 'center',
        backgroundColor: '#DDDDDD',
        paddingRight: 140,
        padding: 25,
        top: 5,
        left: 20,
        bottom: 5
    },
    subjectDelete: {
        position: 'absolute',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#f44141',
        padding: 25,
        top: 5,
        bottom: 5,
        right: 10
    },
    subjectDeleteText: {
        color: 'white'
    },
});
