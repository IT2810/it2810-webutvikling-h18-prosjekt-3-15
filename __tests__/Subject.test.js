import React from 'react';
import Subject from '../components/Subject';
import renderer from 'react-test-renderer';
import Button from "../components/Button";



import ShallowRenderer from 'react-test-renderer/shallow';

it('renders correctly', () => {
    const mockTitle = "Smil";
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Subject subject={mockTitle}/>);
    expect(tree).toMatchSnapshot();
});

test('test Onpress functunality', () =>{
    const mockFunc = jest.fn();

})