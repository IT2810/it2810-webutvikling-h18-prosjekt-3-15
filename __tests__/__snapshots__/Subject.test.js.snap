// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "borderBottomColor": "#ededed",
      "borderBottomWidth": 2,
      "padding": 40,
      "paddingRight": 20,
      "position": "relative",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#DDDDDD",
        "bottom": 5,
        "justifyContent": "center",
        "left": 20,
        "padding": 25,
        "paddingRight": 140,
        "position": "absolute",
        "top": 5,
      }
    }
  >
    <Button
      onPress={[Function]}
      title="Smil"
    >
      Hello
    </Button>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f44141",
        "bottom": 5,
        "justifyContent": "center",
        "padding": 25,
        "position": "absolute",
        "right": 10,
        "top": 5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Delete
    </Text>
  </TouchableOpacity>
</Component>
`;
